File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        # First set a list of strings of vowels
        # Iterate through the string 
        # Sort the vowels in ascending order 
        
        vowels = set('AEIOUaeiou')
        vowel_list = [c for c in s if c in vowels]
        vowel_list.sort()
        
        result = []
        j = 0  
        
        for c in s:
            if c in vowels:
                result.append(vowel_list[j])
                j += 1
            else:
                result.append(c)
        return ''.join(result)

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
        String vowelsStr = "AEIOUaeiou";
        List<Character> vowels = new ArrayList<>();
        char[] arr = s.toCharArray();

        // Extract vowels
        for (char c : arr) {
            if (vowelsStr.indexOf(c) != -1) {
                vowels.add(c);
            }
        }

        // Sort vowels by ASCII value
        Collections.sort(vowels);

        // Replace vowels in original positions with sorted
        int idx = 0;
        for (int i = 0; i < arr.length; i++) {
            if (vowelsStr.indexOf(arr[i]) != -1) {
                arr[i] = vowels.get(idx++);
            }
        }
        return new String(arr);
    }
}
